### PR TITLE
@lennamau: answered questions in answers.md

### PR DESCRIPTION
**Pull request by arunabenji29**

### CODE REVIEW

**Comment by arunabenji29** [comment]
@lennamau : MVP

File: Answers.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+1. What does bcrypt do to slow down attackers?
+
+    bcrypt is used to hash our passwors to make if safe.
+
+1. What are the three parts of the JSON Web Token?
+
+    . Header
+    . Payload
+    . Signature
```
Comment (lennamau) [inline Answers.md:19]: ✔

File: client/src/axiosWithAuth.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export const axiosWithAuth = () => {
+    const token = localStorage.getItem('token')
+
+    return axios.create({
+        headers:{
+            authorization:token
+        }
+    })
+}
```
Comment (lennamau) [inline client/src/axiosWithAuth.js:11]: ✔

File: client/src/components/Login.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    handleSubmit = (event) => {
+        event.preventDefault();
+        axios.post('http://localhost:3300/api/login',this.state)
+        .then(res => {
+            console.log('login data ',res)
+            localStorage.setItem('token',res.data.token)
+            this.props.history.push('/jokes')
+            window.location.reload()
+        })
+        .catch(error => {
```
Comment (lennamau) [inline client/src/components/Login.js:30]: ✔

File: config/db-model.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    return db('users')
+    .where({id})
+    .first()
+}
+
+function findBy(filter){
+    return db('users')
+    .where(filter)
+}
+
```
Comment (lennamau) [inline config/db-model.js:25]: ✔

File: config/routes.js [hunk 5]
```diff
@@ -10,10 +13,44 @@ module.exports = server => {
...
+  Users.add(user)
+  .then(user => {
+    res.status(200).json({
+      message: `Welcome ${user.username} your id is ${user.id}`
+    })
+
+  })
+  .catch(error => {
+    res.status(500).json(error.response)
+  })
```
Comment (lennamau) [inline config/routes.js:29]: ✔

File: config/routes.js [hunk 6]
```diff
@@ -10,10 +13,44 @@ module.exports = server => {
...
+        token
+      })
+    }
+    else {
+      res.status(401).json({message: 'invalid credentials'})
+    }
+  })
+  .catch(error => {
+    res.status(500).json(error)
+  })
```
Comment (lennamau) [inline config/routes.js:53]: ✔

File: config/routes.js [hunk 7]
```diff
@@ -30,3 +67,15 @@ function getJokes(req, res) {
...
+  const payload = {
+    subject:user.id,
+  }
+
+  const options = {
+    expiresIn:'8h'
+  }
+
+  return jwt.sign(payload,secret.jwtSecret,options)
+}
```
Comment (lennamau) [inline config/routes.js:81]: ✔

**Comment by lennamau** [comment]
# Web Auth Challenge
## MVP
- [x]  register in config/routes
- [x]  login config/routes
- [x] JSON Web tokens for auth 
- [x] GET /api/jokes returns a list of jokes

## Stretch
- [x] React app
- [x] React app styled

## What Looks Great
- Great work completing MVP and stretch Aruna!
- Clean, uniform code
- Good file structure
- Good job with frequent commits and descriptive messages
- Styling looks great!

## What Could Be Improved
- Comments in code to describe what it is doing

## Rating (1-3)
### 3

**Pull request merged**